static/js/stream_events.js

```diff
@@ -132,7 +132,7 @@ export function mark_subscribed(sub, subscribers, color) {
 
     // Update unread counts as the new stream in sidebar might
     // need its unread counts re-calculated
-    message_util.do_unread_count_updates(message_list.all.all_messages());
+    message_util.do_unread_count_updates(all_messages_data.all_messages());
```

Comment:
As an aside, this is a great example of why this code sweep is awesome.  We not only break the `message_list` dependency here, but it also highlights some places in the code where we are doing big-hammer things that probably aren't necessary.

I think it's fine to use a big hammer for the redraw, but I don't think we actually need to re-calculate the entire data structure:

~~~
 10 export function do_unread_count_updates(messages) {
 11     unread.process_loaded_messages(messages);
 12     unread_ui.update_unread_counts();
 13     resize.resize_page_components();
 14 }
~~~

Reply:
(We could probably just lift lines 12 and 13 above into `stream_events`.)  There's a similar situation in `stream_muting.js`, where I think we might need to update the actual unread counts data, but we could do it just for the stream without too much modification to `unread.js`.

Reply:
I think this logic to some extent predates the `unread.js` data structure having complete data -- previously we were just doing `get_messages` queries to learn about unread messages, so if there were a stream you'd unsubscribed from and then resubscribed, I'm not sure what'd happen.  Though I think we now remove unreads when you unsubscribe anyway.